### ISSUE TITLE
Source code is different from the describtion in your paper

### ISSUE THREAD

**Issue by jikerWRN** [sshan-zhao/ACMNet#2]
Thanks for your work!
When i read your provided source code, i found that the part of decoder (SGFM) is diffierent from the decsribtion in your paper. The SGFM is described as the following figure:

![1](https://user-images.githubusercontent.com/37199715/96425750-f1d51000-122e-11eb-9a86-430ebc6a8289.png)

But the code implement is different from the describtion in the figure.

        d_ufeat3 = self.d_deconv3(d_feat)  
        r_ufeat3 = self.r_deconv3(r_feat)
        d_gate3 = torch.sigmoid(self.d_gate3(d_ufeat3))
        d_feat = self.d_resblock50(torch.cat([d_feat2, d_ufeat3, d_gate3*r_feat2], 1))
        d_feat = self.d_resblock51(d_feat)
        r_gate3 = torch.sigmoid(self.r_gate3(r_ufeat3))
        r_feat = self.r_resblock50(torch.cat([r_feat2, r_ufeat3, r_gate3*d_feat2], 1))
        r_feat = self.r_resblock51(r_feat)

**Comment by sshan-zhao** [comment]
Could you describe it in detail? Thanks.

**Comment by jikerWRN** [comment]
When I write the reply, I found that I misunderstand it, please forgive my stupidity lol. May i know the reference code of "CoAttnGPBlock" part, such as knn_operation, gather_operation and grouping_operation? where is the source of pointlib? Thanks!

**Comment by sshan-zhao** [comment]
> When I write the reply, I found that I misunderstand it, please forgive my stupidity lol. May i know the reference code of "CoAttnGPBlock" part, such as knn_operation, gather_operation and grouping_operation? where is the source of pointlib? Thanks!

https://github.com/sshan-zhao/ACMNet/tree/master/pointlib
